### PR TITLE
Add trusted types compatibility

### PR DESCRIPTION
**Pull request by fletchto99**
This PR enables turbo clients to route a trusted types policy through to the library. This policy will be called at all dangerous injection sinks. The response object will be passed to the policy, along with the HTML as a string.

### CODE REVIEW

File: src/core/frames/frame_controller.ts [hunk 1]
```diff
@@ -412,11 +420,10 @@ export class FrameController
 
       frame.delegate.fetchResponseLoaded = (fetchResponse: FetchResponse) => {
         if (frame.src) {
-          const { statusCode, redirected } = fetchResponse
-          const responseHTML = frame.ownerDocument.documentElement.outerHTML
-          const response = { statusCode, redirected, responseHTML }
+          const { statusCode, redirected, response } = fetchResponse
+          const visitResponse = { statusCode, redirected, response }
```
Comment (fletchto99) [inline src/core/frames/frame_controller.ts]: I'm not confident this change is consistent with the old logic where the `responseHTML` was pulled from the frame. I attempted to address that in [b248740](https://github.com/github/turbo/pull/1/commits/b24874019287d18707297746dd6876427975ecc7) but it seems to have broken more things than it fixed.

File: src/core/drive/page_snapshot.ts [hunk 2]
```diff
@@ -1,10 +1,20 @@
+import { CSPTrustedTypesPolicy, emptyHTML } from "../../trusted_types"
 import { parseHTMLDocument } from "../../util"
 import { Snapshot } from "../snapshot"
 import { expandURL } from "../url"
 import { HeadSnapshot } from "./head_snapshot"
 
 export class PageSnapshot extends Snapshot<HTMLBodyElement> {
-  static fromHTMLString(html = "") {
+  static fromResponse(response?: Response, html = emptyHTML): PageSnapshot {
+    if (!response || CSPTrustedTypesPolicy == null) {
```
Comment (KyFaSt) [inline src/core/drive/page_snapshot.ts:9]: out of curiosity, when do we expect there to be no response?

File: src/tests/functional/page_snapshot_tests.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import "../helpers/trusted_type_setup"
+import { test } from "@playwright/test"
+import { assert } from "chai"
+
+test.beforeEach(async ({ page }, workerInfo) => {
+  console.log(workerInfo.project.name)
```
Comment (KyFaSt) [inline src/tests/functional/page_snapshot_tests.ts]: did you mean to leave in this console message?

File: src/util.ts [hunk 4]
```diff
@@ -38,6 +39,17 @@ function copyElementAttributes(destinationElement: Element, sourceElement: Eleme
   }
 }
 
+function copyScriptAttributes(destinationElement: HTMLScriptElement, sourceElement: HTMLScriptElement) {
```
Comment (KyFaSt) [inline src/util.ts:42]: should we add a test for createScriptURL too?

**Review by KyFaSt — DISMISSED** [review]
minor nits, great work on this!!

File: src/tests/functional/page_snapshot_tests.ts (hunk 3 above)
Comment (fletchto99) [reply src/tests/functional/page_snapshot_tests.ts]: Whoops, good catch!

File: src/core/drive/page_snapshot.ts (hunk 2 above)
Comment (fletchto99) [reply src/core/drive/page_snapshot.ts:9]: https://github.com/github/turbo/blob/b013ad76c317515e6b4a6cbc75673d757902e376/src/core/drive/visit.ts#L267

In this case it's possible for the responseHTML to be null

File: src/util.ts (hunk 4 above)
Comment (fletchto99) [reply src/util.ts:42]: I think we actually get that for free from the trusted types test suite.

**Review by manuelpuyol — APPROVED** [review]

**Pull request merged**